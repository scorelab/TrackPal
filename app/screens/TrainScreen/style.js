import { StyleSheet, Dimensions } from "react-native";
let deviceWidth = Dimensions.get("window").width;
let deveiceHeight = Dimensions.get("window").height;

export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "flex-start",
    alignItems: "center",
    backgroundColor: "#fff",
    width: "100%"
  },
  scrollView: {
    width: "100%"
  },
  scrollViewContent: {
    alignItems: "center",
    paddingBottom: 10
  },
  iconStyle: {
    position: "absolute",
    marginTop: "150%",
    paddingLeft: "75%"
  },
  input: {
    height: 45,
    width: deviceWidth * 0.6,
    borderRadius: 5,
    backgroundColor: "rgba(0,0,0,0.1)",
    marginBottom: 15,
    color: "black",
    paddingHorizontal: 10
  },
  searchFilter: {
    backgroundColor: "#3c5898",
    height: 50,
    width: 150,
    alignItems: "center",
    marginLeft: 50,
    borderRadius: 50
  },
  filterText: {
    color: "white",
    marginTop: 12,
    fontSize: 20
  },
  feedText: {
    fontSize: 30,
    marginRight: deviceWidth * 0.6,
    padding: 10,
    fontWeight: "bold",
    color: "black"
  },
  shareLocationTouchableScale: {
    position: "absolute",
    marginTop: deveiceHeight*0.8,
    zIndex: 1
  },
  shareLocationImage:{
    height: 70,
    width: 70,
    backgroundColor: "rgba(0,0,0,0.05)",
    padding: 10,
    borderRadius: 50
  }
});
