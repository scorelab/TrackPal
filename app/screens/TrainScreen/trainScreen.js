import React, { Component } from "react";
import {
  View,
  ScrollView,
  Text,
  ActivityIndicator,
  KeyboardAvoidingView,
  TextInput,
  TouchableOpacity,
  Image
} from "react-native";
import SearchHeaderBar from "../../components/SearchHeaderbar/searchHeaderbar.js";
import ListItems from "../../components/ListComponent/listItemComponent.js";
import styles from "./style.js";
import TouchbleScale from "react-native-touchable-scale";
import Modal from "react-native-modal";

import { f, database } from "../../../config/config.js";

export default class TrainScreen extends Component {
  constructor(props) {
    super(props);

    this.state = {
      objects: null,
      indicator: true,
      currentUserID: null,
      isModalVisible: false,
      filterKey: ""
    };
  }

  componentDidMount() {
    var that = this;
    database.ref("/TrainDetails").on("value", function(snapshot) {
      that.setState({
        objects: snapshotToArray(snapshot),
        indicator: false
      });
    });

    var user = f.auth().currentUser;
    if (user !== null) {
      this.setState({
        currentUserID: user.uid
      });
    }
  }

  redirectUserToSharingScreen(objectKey) {
    var currentUser = f.auth().currentUser;

    if (currentUser === null) {
      alert("Sorry!, You are not a registered User! Please Register First.");
      this.props.navigation.navigate("Login");
    } else {
      this.props.navigation.navigate("SharingLocation", {
        userKey: objectKey,
        prevScreen: "TrainScreen"
      });
    }
  }

  searchByTheDestination(destination) {
    if (destination !== null || destination !== "") {
      var that = this;
      this.setState({
        isModalVisible: false
      });
      this.setState({ indicator: true });

      database
        .ref("/TrainDetails")
        .orderByChild("destination")
        .equalTo(destination)
        .on("value", function(snapshot) {
          that.setState({
            objects: snapshotToArray(snapshot),
            indicator: false
          });
        });
    } else {
      this.setState({
        isModalVisible: false
      });
    }
  }

  toggleModal = () => {
    this.setState({ isModalVisible: !this.state.isModalVisible });
  };

  render() {
    const { currentUserID } = this.state;
    if (this.state.indicator) {
      return (
        <View style={styles.container}>
          <SearchHeaderBar />
          <Text style={styles.feedText}>Your feed</Text>
          <ScrollView
            style={styles.scrollView}
            contentContainerStyle={styles.scrollViewContent}
            showsVerticalScrollIndicator={false}
          >
            <View style={[styles.container, styles.horizontal]}>
              <ActivityIndicator size="large" color="#0000ff" />
            </View>
          </ScrollView>
          <TouchbleScale
            style={styles.shareLocationTouchableScale}
            onPress={() =>
              this.props.navigation.navigate("ShareLocation", {
                prevScreen: "TrainScreen"
              })
            }
          >
            <View elevation={5}>
              <Image
                source={require("../../images/updated_logo.png")}
                style={styles.shareLocationImage}
              />
            </View>
          </TouchbleScale>
        </View>
      );
    } else {
      return (
        <View style={styles.container}>
          {/* add model to search the list with destination function */}
          <Modal
            isVisible={this.state.isModalVisible}
            onBackdropPress={this.toggleModal}
          >
            <View
              style={{
                backgroundColor: "white",
                margin: 10,
                padding: 50,
                borderRadius: 15
              }}
            >
              <Text>Enter the Detination Name</Text>
              <KeyboardAvoidingView behavior="position">
                <TextInput
                  placeholder="Destination"
                  placeholderTextColor="rgba(0,0,0,0.5)"
                  style={styles.input}
                  onChangeText={text => this.setState({ filterKey: text })}
                />
                <TouchableOpacity
                  style={styles.searchFilter}
                  onPress={() =>
                    this.searchByTheDestination(this.state.filterKey)
                  }
                >
                  <Text style={styles.filterText}>Search</Text>
                </TouchableOpacity>
              </KeyboardAvoidingView>
            </View>
          </Modal>

          <SearchHeaderBar />
          <Text style={styles.feedText}>Your feed</Text>
          <ScrollView
            style={styles.scrollView}
            contentContainerStyle={styles.scrollViewContent}
            showsVerticalScrollIndicator={false}
          >
            {this.state.objects.map((objects, key) => {
              if (objects.finished === false) {
                if (objects.key === currentUserID) {
                  return (
                    <ListItems
                      from={objects.from.title2}
                      to={objects.to.title1}
                      userID={objects.key}
                      screen="train_"
                      posted={objects.trainName}
                      userImage={require("../../images/train/traine.jpg")}
                      message="This is your shared details. Tap to view"
                      onPress={() =>
                        this.redirectUserToSharingScreen(objects.key)
                      }
                      key={key}
                    />
                  );
                } else {
                  return (
                    <ListItems
                      from={objects.from.title2}
                      to={objects.to.title1}
                      userID={objects.key}
                      screen="train_"
                      posted={objects.trainName}
                      userImage={require("../../images/train/traine.jpg")}
                      message="Posted by someone else. Tap to view location sharing and shared person's details."
                      onPress={() =>
                        this.redirectUserToSharingScreen(objects.key)
                      }
                      key={key}
                    />
                  );
                }
              }
            })}
          </ScrollView>
          <TouchbleScale
            style={styles.shareLocationTouchableScale}
            onPress={() =>
              this.props.navigation.navigate("ShareLocation", {
                prevScreen: "TrainScreen"
              })
            }
          >
            <View elevation={5}>
              <Image
                source={require("../../images/updated_logo.png")}
                style={styles.shareLocationImage}
              />
            </View>
          </TouchbleScale>
          {/* <Icon
            name="search"
            reverse
            raised
            color="blue"
            containerStyle={{
              position: "absolute",
              marginTop: "150%",
              paddingRight: "60%"
            }}
            onPress={this.toggleModal}
          /> */}
        </View>
      );
    }
  }
}

// method to conver the json object to js array
function snapshotToArray(snapshot) {
  var returnArr = [];

  snapshot.forEach(function(childSnapshot) {
    var item = childSnapshot.val();
    item.key = childSnapshot.key;

    returnArr.push(item);
  });

  return returnArr;
}
