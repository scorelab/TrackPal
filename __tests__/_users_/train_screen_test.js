import React from 'react';
import TrainScreen from '../../app/screens/TrainScreen/trainScreen';

import renderer from 'react-test-renderer';

jest.useFakeTimers();

test('renders correctly', () => {
  const tree = renderer.create(<TrainScreen/>).toJSON();
  expect(tree).toMatchSnapshot();
});