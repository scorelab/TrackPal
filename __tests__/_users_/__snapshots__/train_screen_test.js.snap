// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00aced",
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "width": 375,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
          }
        }
      >
        Track Your Train
      </Text>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "paddingBottom": 10,
      }
    }
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "backgroundColor": "#e8e8e8",
        "width": "100%",
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "flex": 1,
              "justifyContent": "flex-start",
              "width": "100%",
            },
            undefined,
          ]
        }
      >
        <ActivityIndicator
          animating={true}
          color="#0000ff"
          hidesWhenStopped={true}
          size="large"
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "marginTop": "150%",
        "paddingLeft": "75%",
        "position": "absolute",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderRadius": 28,
          "height": 52,
          "justifyContent": "center",
          "margin": 7,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 52,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 24,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="iconIcon"
      >
        î……
      </Text>
    </View>
  </View>
</View>
`;
